consensus/istanbul/backend/engine.go

```diff
@@ -366,10 +365,11 @@ func (sb *Backend) snapshot(chain consensus.ChainHeaderReader, number uint64, ha
 			}
 
 			var validators []common.Address
-			if sb.config.ValidatorContract != (common.Address{}) && sb.config.GetValidatorSelectionMode(big.NewInt(0)) == params.ContractMode {
-				sb.logger.Info("Initialising snap with contract validators", "address", sb.config.ValidatorContract, "client", sb.config.Client)
+			var targetBlockNumber = big.NewInt(0)
+			if sb.config.GetValidatorContractAddress(targetBlockNumber) != (common.Address{}) && sb.config.GetValidatorSelectionMode(targetBlockNumber) == params.ContractMode {
```

Comment:
saving sb.config.GetValidatorContractAddress(targetBlockNumber) in local variable can save to execute same @ l.372